pages.fa/common/grep.md

```diff
@@ -0,0 +1,32 @@
+# grep
+
+> یافتن الگو در فایل ها به کمک عبارات با قاعده
+> اطلاعات بیشتر : <https://www.gnu.org/software/grep/manual/grep.html>.
+
+- جستجو یک الگو در یک فایل :
+
+`grep "{{الگو_جستجو}}" {{مسیر/فایل}}`
+
+- جستجو یک عبارت خاص (معادل مقایسه رشته ای) :
+
+`grep --fixed-strings "{{رشته_مورد_نظر}}" {{مسیر/فایل}}`
+
+- جستجو بازگشتی یک الگو در تمامی فایل های یک پوشه، نمایش تمامی خطوط منطبق، فایل های باینری را رد میکند
+
+`grep --recursive --line-number --binary-files={{without-match}} "{{الگو_جستجو}}" {{path/to/directory}}`
+
+- استفاده از عبارات با قاعده توسعه یافته (با پشتیبانی از `?`، `+`، `{}`، `()` و `|`)، در حالت حساس به بزرگی کوچکی کاراکتر ها :
+
+`grep --extended-regexp --ignore-case "{{الگو_جستجو}}" {{مسیر/فایل}}`
+
+- چاپ 3 خط از قبل و بعد محل انطباق
+
+`grep --{{context|before-context|after-context}}={{3}} "{{الگو_جستجو}}" {{مسیر/فایل}}`
+
+- چاپ نام فایل و شماره خط برای هر انطباق با رنگبندی :
+
+`grep --with-filename --line-number --color=always "{{الگو_جستجو}}" {{مسیر/فایل}}`
+
+- جستجوی خطوط منطبق، چاپ متن منطبق :
+
+`grep --only-matching "{{الگو_جستجو}}" {{مسیر/فایل}}`
```

Comment:
(Optional) We recently added a new command to the last of the `grep` page.

```
 - Search `stdin` for lines that do not match a pattern:

`cat path/to/file | grep --invert-match "search_pattern"`
```

Can you translate this too?